cloud-service-provider/ibm/deployable-architectures/productivity-suite/README.md

```diff
@@ -0,0 +1,347 @@
+   - **ChatQnA**: Conversational AI with RAG capabilities
+   - **DocSum**: Document summarization
+   - **CodeGen**: Code generation assistance
+4. **Shared Infrastructure**:
+   - Keycloak for authentication
+   - Vector databases for document retrieval
+   - MongoDB for data persistence
+
+Each megaservice comprises multiple specialized microservices:
+
+### ChatQnA Components
+
+- LLM Service
+- Embedding Service
+- Retriever Service
+- Reranking Service
+
+### DocSum Components
+
+- Document Processing
+- Summarization Service
+
+### CodeGen Components
+
+- Code Understanding
+- LLM Code Generation
+
+## Modules
+
+The solution includes the following Terraform modules:
+
+- `ibm-cloud-vpc`: Provisions the IBM Cloud VPC infrastructure and IKS cluster
+- `chatqna`: Deploys the ChatQnA megaservice
+- `codegen`: Deploys the CodeGen megaservice
+- `docsum`: Deploys the DocSum megaservice
+- 'chathistory': Deploys the chathistory service
+- 'prompt': Deploys the prompt registry service
+
+Supporting infrastructure:
+
+- Keycloak: Identity and access management
+- Nginx: Central gateway for all services
+- UI: Unified user interface
+
+## Resource Requirements
+
+The infrastructure provisioning creates IBM Cloud VPC resources with the following specifications:
+
+- **Compute**: Intel-based virtual servers (recommended: 32+ vCPU cores)
+- **Memory**: Minimum 64GB RAM for full deployment
+- **Storage**: 100GB+ for model storage
+- **Networking**: VPC with public and private subnets
+- **Kubernetes**: IBM Kubernetes Service (IKS) cluster with worker nodes provisioned in the VPC
+
+## Inputs
+
+### Core Infrastructure Variables
+
+| Name                                        | Description                            | Type   | Required |
+| ------------------------------------------- | -------------------------------------- | ------ | :------: |
+| ibmcloud_api_key                            | IBM Cloud API key                      | string |   yes    |
+| HuggingFace Token                           | HF Token with access to desired models | string |   yes    |
+| Intel® AI for Enterprise Inference API Key | API Keys for LLM endpoints             | string |   yes    |
+| email                                       | IBM Cloud email address                | string |    no    |
+| region                                      | IBM Cloud region                       | string |   yes    |
+| vpc_name                                    | Name of the VPC to create              | string |   yes    |
+| cluster_name                                | Name of the IKS cluster                | string |   yes    |
+| worker_count                                | Number of worker nodes                 | number |   yes    |
+| machine_type                                | Worker node instance profile           | string |   yes    |
+| cluster_version                             | Kubernetes version for IKS cluster     | string |   yes    |
+
+### Application Variables
```

Comment:
In the table below, use for all helm chart references (ChatQnA is one example)

chatqna_helm_chart_path - > chatqna_helm_chart 
Path to the ChatQNA Helm chart -> ChatQNA Helm chart repository